File: Server/src/main/java/core/game/system/mysql/SQLManager.java
package core.game.system.mysql;

import rs09.ServerConstants;
import rs09.game.system.SystemLogger;
import core.game.system.SystemManager;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

/**
 * Manages the SQL connections.
 * @author Vexia
 * 
 */
public final class SQLManager {
	
	/**
	 * If the sql manager is locally hosted. Generally you never should change this. Ignore this.
	 */
	public static final boolean LOCAL = false;

    /**
     * The database URL.
     */
    public static String DATABASE_URL;

	/**
	 * IF the sql manager is initialized.
	 */
	private static boolean initialized;

	/**
	 * Constructs a new {@code SQLManager} {@code Object}
	 */
	public SQLManager() {
		/**
		 * empty.
		 */
	}

	/**
	 * Initializes the sql manager.
	 */
	public static void init() {
		initialized = true;
		SystemManager.getSystemConfig().parse();
	}
	
	/**
	 * Pre-plugin parsing.
	 */
	public static void prePlugin() {
			//new NPCConfigSQLHandler().parse();

	}

	/**
	 * Parses data from the database for the server post plugin loading.
	 */
	public static void postPlugin() {

	}

	/**
	 * Gets a connection.
	 * @return The connection.
	 */
	public static Connection getConnection() {
		DATABASE_URL  = ServerConstants.DATABASE_ADDRESS + ":" + ServerConstants.DATABASE_PORT + "/" + ServerConstants.DATABASE_NAME + "?useTimezone=true&serverTimezone=UTC";
		try {
			return DriverManager.getConnection("jdbc:mysql://" +   DATABASE_URL, ServerConstants.DATABASE_USER, ServerConstants.DATABASE_PASS);
		} catch (SQLException e) {
			SystemLogger.logErr("Mysql error message=" + e.getMessage() + ".");
		}
		return null;
	}

	/**
	 * Releases the connection so it's available for usage.
	 * @param connection The connection.
	 */
	public static void close(Connection connection) {
		try {
			connection.close();
		} catch (SQLException e) {
			e.printStackTrace();
		}
	}

	/**
	 * Gets the initialized.
	 * @return the initialized
	 */
	public static boolean isInitialized() {
		return initialized;
	}

	/**
	 * Sets the bainitialized.
	 * @param initialized the initialized to set.
	 */
	public static void setInitialized(boolean initialized) {
		SQLManager.initialized = initialized;
	}

}

File: Management-Server/src/main/java/ms/system/mysql/SQLManager.kt
package ms.system.mysql

import ms.system.util.ManagementConstants
import ms.system.util.ManagementConstants.DATABASE_HOST_ADDRESS
import ms.system.util.ManagementConstants.DATABASE_NAME
import ms.system.util.ManagementConstants.DATABASE_PASSWORD
import ms.system.util.ManagementConstants.DATABASE_PORT
import ms.system.util.ManagementConstants.DATABASE_USERNAME
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

/**
 * Manages the sql connections.
 * @author Vexia
 */
object SQLManager {

    /**
     * IF the sql manager is initialized.
     */
    var isInitialized = false

    /**
     * Initializes the sql manager.
     */
    @JvmStatic
    fun init() {
        isInitialized = true
        WorldListSQLHandler.clearWorldList()
    }

    /**
     * Gets a connection from the pool.
     * @return The connection.
     */
    @JvmStatic
    val connection: Connection?
        get() {
            try {
                return DriverManager.getConnection(
                    "jdbc:mysql://$DATABASE_HOST_ADDRESS:$DATABASE_PORT/$DATABASE_NAME?useTimezone=true&serverTimezone=UTC",
                    DATABASE_USERNAME,
                    DATABASE_PASSWORD
                )
            } catch (e: SQLException) {
                println("Error: Mysql error message=" + e.message + ".")
            }
            return null
        }

    /**
     * Releases the connection so it's available for usage.
     * @param connection The connection.
     */
    @JvmStatic
    fun close(connection: Connection) {
        try {
            connection.close()
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }
}
